### PR TITLE
Udash 0.5

### PR DESCRIPTION
**Pull request by Starzu**

### CODE REVIEW

File: backend/src/main/scala/io/udash/web/guide/rest/HttpSerializationUtils.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import akka.http.scaladsl.marshalling.Marshaller
+import akka.http.scaladsl.model.HttpEntity.Strict
+import akka.http.scaladsl.model.{HttpEntity, MediaTypes}
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import com.avsystem.commons.serialization.GenCodec
+import io.udash.rpc.DefaultUdashSerialization
+
+trait HttpSerializationUtils extends DefaultUdashSerialization {
+//  implicit def optionMarshaller[T](implicit codec: GenCodec[T]) =
```
Comment (ddworak) [inline backend/src/main/scala/io/udash/web/guide/rest/HttpSerializationUtils.scala]: Please don't.

File: shared/src/main/scala/io/udash/web/guide/GuideExceptions.scala [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import com.avsystem.commons.serialization.GenCodec
+import io.udash.i18n.TranslationKey0
+import io.udash.rpc.serialization.{DefaultExceptionCodecRegistry, ExceptionCodecRegistry}
+
+object GuideExceptions {
+  case class ExampleException(msg: String) extends Exception(msg)
+  case class TranslatableExampleException(trKey: TranslationKey0) extends Exception
+
+  val registry: ExceptionCodecRegistry = {
+    val registry = new DefaultExceptionCodecRegistry
```
Comment (ddworak) [inline shared/src/main/scala/io/udash/web/guide/GuideExceptions.scala]: Use `.setup`

File: shared/src/main/scala/io/udash/web/guide/demos/i18n/Translations.scala [hunk 3]
```diff
@@ -20,4 +20,8 @@ object Translations {
       val buttonLabel = key("auth.register.buttonLabel")
     }
   }
+
+  object exceptions {
```
Comment (ddworak) [inline shared/src/main/scala/io/udash/web/guide/demos/i18n/Translations.scala:24]: I'd rather these started with an uppercase.

File: backend/src/main/scala/io/udash/web/guide/rest/DevsGuideRest.scala [hunk 4]
```diff
@@ -1,14 +1,22 @@
...
-trait DevsGuideRest extends HttpServiceBase with SpraySerializationUtils {
+object DevsGuideRest extends HttpSerializationUtils {
+  import akka.http.scaladsl.server.Directives._
 
-  val route = {
+  private implicit val system = ActorSystem()
+  private implicit val materializer = ActorMaterializer()
+  private implicit val executionContext = system.dispatcher
+
+  private val route = CorsDirectives.cors() {
```
Comment (ddworak) [inline backend/src/main/scala/io/udash/web/guide/rest/DevsGuideRest.scala:19]: We might not need this ;)

Comment (Starzu) [reply backend/src/main/scala/io/udash/web/guide/rest/DevsGuideRest.scala:19]: REST API works on different port.

**Review by ddworak — APPROVED** [review]

**Pull request merged**